### ISSUE TITLE
MotiLink not working with App Router

### ISSUE THREAD

**Issue by lolo-io** [nandorojo/solito#466]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Do you want this issue prioritized?

- [ ] Yes, I have sponsored
- [ ] Not urgent

### Current Behavior

I am importing MotiLink from "solito/moti/app"

Whenever I add a <MotiLink> component in one of my Client Component, I get this error :

Unhandled Runtime Error
NotFoundError: Node.removeChild: The node to be removed is not a child of this node

### Expected Behavior

Link should work

### Versions

```markdown
- Solito: 4.2.1
- Next.js: 14.1.0
- Expo: error is on web
- React Native: 0.71.3
```

### Screenshots

![image](https://github.com/nandorojo/solito/assets/1267636/dcef2aed-8a75-48b3-8a9c-2376e4b9075b)

**Comment by nandorojo** [comment]
There’s zero code here so I can’t help. Maybe you’re importing from the wrong place

**Comment by nandorojo** [comment]
The error points to safe area context, not solito or moti. What makes you think these are related exactly

**Comment by lolo-io** [comment]
What made me think that,  was that when I removed `MotiLink` from my component, everything worked fine.

But I found a solution now. 
I am in the process of transitioning from next page router to app router and I am using React Native Paper for styling, I noticed that removing `<PaperProvider/>` led me to another error:` "setImmediate is not defined"`. 

I added `import 'setimmediate'` to my `index.ts`  file where my providers are stacked, and kept  `<PaperProvider/>`  and everything works fine now.

Adding  `import 'setimmediate'` to the layout file didn't work.

So for me if anyone encounters a similar problem try adding `import 'setimmediate'` to your index.ts file.
